source/server/spongineer/debugging.rst

```diff
@@ -119,3 +119,104 @@ A ``NoClassDefFoundError`` occurs when the plugin tries to access a class that i
+    /*     net.minecraft.world.World.func_72870_g(World.java:1952)                                                                */
+    /*     net.minecraft.world.World.func_72939_s(World.java:6596)                                                                */
+    /*     net.minecraft.world.WorldServer.func_72939_s(WorldServer.java:2300)                                                    */
+    /*     net.minecraft.server.MinecraftServer.func_71190_q(MinecraftServer.java:767)                                            */
+    /*     net.minecraft.server.dedicated.DedicatedServer.func_71190_q(DedicatedServer.java:396)                                  */
+    /*     net.minecraft.server.MinecraftServer.func_71217_p(MinecraftServer.java:668)                                            */
+    /*     net.minecraft.server.MinecraftServer.run(MinecraftServer.java:526)                                                     */
+    /*     java.lang.Thread.run(Unknown Source)                                                                                   */
+    /*  Phases Remaining:                                                                                                         */
+    /*                                                                                                                            */
+    /* Minecraft : 1.12.2                                                                                                         */
+    /* SpongeAPI : 7.1.0-SNAPSHOT-7105dfc                                                                                         */
+    /* SpongeForge : 1.12.2-2705-7.1.0-BETA-3361                                                                                  */
+    /* Minecraft Forge : 14.23.4.2705                                                                                             */
+    /******************************************************************************************************************************/
+
+This stacktrace contains the most important version numbers, as well some information about the phase the server was in.
+In this case a ``NullPointerException`` has been thrown during ``EntityTickPhase``. At this point it's important to
+check which plugins are involved with the crash. This requires some research as you have to match the package names with
+mod names; checking the ``Caused by`` blocks may also help.
+
+* ``java`` classes can be ignored during the error search.
+* ``net.minecraft`` is the vanilla Minecraft code. If only these elements are present, it's either a Minecraft bug or a
+  really nasty one (these are explained later).
+* ``org.spongepowered`` is from Sponge itself, having only these and Minecraft packages present usually indicates a
+  Sponge bug (or a nasty one). The :doc:`bugreport` chapter explains where to report bugs.
```

Comment:
can we find a better name for this? nasty bug sounds like an STD to me, maybe "or another coremod being present"